frontend/src/components/file/AdvancedFileUpload.vue

```diff
@@ -23,17 +23,37 @@ const {
   disabled?: boolean;
 }>();
 
+const emit = defineEmits<{
+  (event: 'activity-generated', activityId: string, submissionId: string): void;
+}>();
```

Comment:
Emits in the script section should be in camel case for convention, so it should be `activityGenerated` both here and later in line 46. Vue will automatically detect it in kebab case later on in line 794 and trigger there even when it's `activity-generated` there.